packages/manager/src/features/CloudPulse/Utils/FilterBuilder.ts

```diff
@@ -266,9 +297,14 @@ export const checkIfWeNeedToDisableFilterByFilterKey = (
     if (filter) {
       return filter.configuration.dependency?.some((dependent) => {
         const dependentFilter = dependentFilters[dependent];
+        const optionalFilter = filters.find(
```

Comment:
here time complexity is not good, for all dependency we will again iterate, so better to put this outside...and store list of optional filters and not care about it